### ISSUE TITLE
Cannot upgrade to v0.4.0 because of workqueueMetricsProvider 

### ISSUE THREAD

**Issue by LouisPlisso** [kubernetes-sigs/controller-runtime#727]
Tried to upgrade from v0.3.0 to v0.4.0, resulting in this error:

```
(default) [louis.plissonneau@Louiss-MacBook-Pro:…/10gen/ops-manager-kubernetes] cat go.mod
module github.com/10gen/ops-manager-kubernetes

require (
        github.com/beorn7/perks v1.0.1 // indirect
        github.com/blang/semver v3.5.1+incompatible
        github.com/emicklei/go-restful v2.9.6+incompatible // indirect
        github.com/go-openapi/jsonreference v0.19.3 // indirect
        github.com/go-openapi/spec v0.19.4 // indirect
        github.com/google/uuid v1.1.1
        github.com/hashicorp/golang-lru v0.5.3 // indirect
        github.com/imdario/mergo v0.3.8
        github.com/json-iterator/go v1.1.8 // indirect
        github.com/pkg/errors v0.8.1
        github.com/prometheus/client_golang v0.9.4 // indirect
        github.com/prometheus/procfs v0.0.8 // indirect
        github.com/spf13/cast v1.3.0
        github.com/spf13/pflag v1.0.5 // indirect
        github.com/stretchr/testify v1.4.0
        github.com/xdg/stringprep v1.0.0
        go.uber.org/zap v1.10.0
        gopkg.in/yaml.v2 v2.2.7 // indirect
        k8s.io/api v0.0.0-20191114100237-2cd11237263f // 1.15.6
        k8s.io/apimachinery v0.0.0-20191004115701-31ade1b30762 // 1.15.6
        k8s.io/client-go v0.0.0-20191114101336-8cba805ad12d // 1.15.6
        k8s.io/code-generator v0.0.0-20190912054826-cd179ad6a269 // 1.15.6
        sigs.k8s.io/controller-runtime v0.4.0
)

go 1.13
(default) [louis.plissonneau@Louiss-MacBook-Pro:…/10gen/ops-manager-kubernetes] GOFLAGS="" go get -u=patch
# sigs.k8s.io/controller-runtime/pkg/metrics
../../../../pkg/mod/sigs.k8s.io/controller-runtime@v0.4.0/pkg/metrics/workqueue.go:32:48: cannot use composite literal (type workqueueMetricsProvider) as type workqueue.MetricsProvider in argument to workqueue.SetProvider:
        workqueueMetricsProvider does not implement workqueue.MetricsProvider (missing NewDeprecatedAddsMetric method)
```

Error message is similar to #335

**Comment by LouisPlisso** [comment]
@DirectXMan12 created this new issue instead of following on the previous one

**Comment by kfyharukz** [comment]
Maybe, v0.4.0 depends on k8s 1.16 client libraries.
I could remove the error with the following go.mod.
```
module github.com/cybozu-go/topolvm

go 1.12

replace launchpad.net/gocheck => github.com/go-check/check v0.0.0-20180628173108-788fd7840127

require (
	github.com/GeertJohan/go.rice v1.0.0 // indirect
	github.com/cloudflare/cfssl v0.0.0-20190510060611-9c027c93ba9e
	github.com/container-storage-interface/spec v1.1.0
	github.com/cybozu-go/log v1.5.0
	github.com/cybozu-go/well v1.9.0
	github.com/go-logr/logr v0.1.0
	github.com/go-logr/zapr v0.1.1 // indirect
	github.com/go-sql-driver/mysql v1.4.1 // indirect
	github.com/golang/protobuf v1.3.2
	github.com/google/certificate-transparency-go v1.0.21 // indirect
	github.com/imdario/mergo v0.3.7 // indirect
	github.com/jmhodges/clock v0.0.0-20160418191101-880ee4c33548 // indirect
	github.com/jmoiron/sqlx v1.2.0 // indirect
	github.com/kisielk/sqlstruct v0.0.0-20150923205031-648daed35d49 // indirect
	github.com/kubernetes-csi/csi-test v2.2.0+incompatible
	github.com/lib/pq v1.1.1 // indirect
	github.com/mattn/go-sqlite3 v1.10.0 // indirect
	github.com/onsi/ginkgo v1.8.0
	github.com/onsi/gomega v1.5.0
	github.com/prometheus/client_golang v1.1.0
	github.com/prometheus/client_model v0.0.0-20190129233127-fd36f4220a90
	github.com/prometheus/common v0.6.0
	github.com/spf13/cobra v0.0.5
	github.com/spf13/viper v1.3.2
	go.uber.org/atomic v1.4.0 // indirect
	go.uber.org/zap v1.10.0 // indirect
	golang.org/x/sys v0.0.0-20190801041406-cbf593c0f2f3
	golang.org/x/time v0.0.0-20190308202827-9d24e82272b4 // indirect
	google.golang.org/grpc v1.23.1
	k8s.io/api v0.16.4
	k8s.io/apiextensions-apiserver v0.0.0-20190918201827-3de75813f604 // indirect
	k8s.io/apimachinery v0.16.4
	k8s.io/client-go v0.16.4
	k8s.io/klog v0.4.0
	sigs.k8s.io/controller-runtime v0.4.0
)
```

**Comment by LouisPlisso** [comment]
@kfyharukz thanks for the investigation
indeed on k8s 1.16 I don't get the error

I wasn't expecting the bump to 1.16 to  break this

**Comment by LouisPlisso** [comment]
I see again the same issue with v0.5.2 with operator-sdk v0.17.0 [using k8s 1.17.5]

Do you understand how this could manifest with k8s 1.17.5???

```
# sigs.k8s.io/controller-runtime/pkg/metrics
../../../../pkg/mod/sigs.k8s.io/controller-runtime@v0.5.2/pkg/metrics/workqueue.go:99:48: cannot use workqueueMetricsProvider literal (type workqueueMetricsProvider) as type workqueue.MetricsProvider in argument to workqueue.SetProvider:
        workqueueMetricsProvider does not implement workqueue.MetricsProvider (missing NewDeprecatedAddsMetric method)
```


here is my go.mod file:
```
module github.com/10gen/ops-manager-kubernetes

require (
	github.com/beorn7/perks v1.0.1 // indirect
	github.com/blang/semver v3.5.1+incompatible
	github.com/emicklei/go-restful v2.9.6+incompatible // indirect
	github.com/evanphx/json-patch v4.5.0+incompatible
	github.com/go-openapi/jsonreference v0.19.3 // indirect
	github.com/go-openapi/spec v0.19.7 // indirect
	github.com/go-openapi/swag v0.19.9 // indirect
	github.com/golang/protobuf v1.3.5 // indirect
	github.com/google/uuid v1.1.1
	github.com/hashicorp/go-multierror v1.0.0
	github.com/hashicorp/golang-lru v0.5.4 // indirect
	github.com/imdario/mergo v0.3.9
	github.com/json-iterator/go v1.1.9 // indirect
	github.com/mailru/easyjson v0.7.1 // indirect
	github.com/operator-framework/operator-sdk v0.17.0
	github.com/pkg/errors v0.9.1
	github.com/prometheus/procfs v0.0.11 // indirect
	github.com/spf13/cast v1.3.1
	github.com/stretchr/testify v1.4.0
	github.com/xdg/stringprep v1.0.0
	go.uber.org/zap v1.14.1
	k8s.io/api v0.17.5
	k8s.io/apiextensions-apiserver v0.17.5 // indirect
	k8s.io/apimachinery v0.17.5
	k8s.io/client-go v12.0.0+incompatible
	k8s.io/code-generator v0.17.5
	sigs.k8s.io/controller-runtime v0.5.2
	sigs.k8s.io/testing_frameworks v0.1.2 // indirect
)

go 1.13

```

**Comment by alexeldeib** [comment]
This is usually a dependency mismatch between k8s libraries.

> k8s.io/client-go v12.0.0+incompatible

This does not match your other versions, it's using the older semantic versioning. Try aligning the version here to v0.17.5

**Comment by LouisPlisso** [comment]
@alexeldeib thanks, but this is automatically replaced by go get even if I put v0.17.5 here

I have a fix that allows controller-runtime v0.5.2 to work with operator sdk v0.17.0

**Comment by LouisPlisso** [comment]
adding a replace to go.mod solves this better ;)

```
replace k8s.io/client-go => k8s.io/client-go v0.17.5
```

**Comment by alexeldeib** [comment]
I see, I didn't realize this was an OSDK specific issue. I was able to repro this and I'm not using OSDK; matching my versions fixed it.

That's a bit curious -- doesn't it mean OSDK is using some older client-go features?
